### PR TITLE
Added macos as -os option

### PR DESCRIPTION
**Pull request by maridematte**
Issue https://github.com/dotnet/install-scripts/issues/406

Added the option for `macos` for `-os`. 

When `macos` is passed as an option, it is converted to `osx` so the download links still work.

### CODE REVIEW

**Review by baronfel — APPROVED** [review]
Thanks!

File: src/dotnet-install.sh [hunk 1]
```diff
@@ -423,8 +423,13 @@ get_normalized_os() {
...
                 return 0
                 ;;
+            macos)
+                osname='osx'
+                echo "$osname"
+                return 0
+                ;;
             *)
-                say_err "'$user_defined_os' is not a supported value for --os option, supported values are: osx, linux, linux-musl, freebsd, rhel.6. If you think this is a bug, report it at https://github.com/dotnet/install-scripts/issues."
+                say_err "'$user_defined_os' is not a supported value for --os option, supported values are: osx, macos, linux, linux-musl, freebsd, rhel.6. If you think this is a bug, report it at https://github.com/dotnet/install-scripts/issues."
```
Comment (danmoseley) [inline src/dotnet-install.sh:432]:
```suggestion
                say_err "'$user_defined_os' is not a supported value for --os option, supported values are: osx, macos, linux, linux-musl, freebsd, rhel.6. If you think this is a bug, we would welcome a report in our repo at https://github.com/dotnet/install-scripts/issues."
```

Comment (danmoseley) [reply src/dotnet-install.sh:432]: Feel free to ignore! I am all about being friendly!

File: tests/Install-Scripts.Test/GivenThatIWantToGetTheSdkLinksFromAScript.cs [hunk 2]
```diff
@@ -371,6 +371,29 @@ public async Task WhenAnExactVersionIsPassedToBash(string version, string runtim
...
+                    "--os", "macos",
+                    "-dryrun" };
+
+            var commandResult = CreateInstallCommand(args)
+                            .CaptureStdOut()
+                            .CaptureStdErr()
+                            .Execute();
+
+            commandResult.Should().Pass();
+            commandResult.Should().NotHaveStdErr();
```
Comment (danmoseley) [inline tests/Install-Scripts.Test/GivenThatIWantToGetTheSdkLinksFromAScript.cs:393]: Verify it used osx?

**Review by danmoseley — APPROVED** [review]

File: tests/Install-Scripts.Test/GivenThatIWantToGetTheSdkLinksFromAScript.cs (hunk 2 above)
Comment (danmoseley) [reply tests/Install-Scripts.Test/GivenThatIWantToGetTheSdkLinksFromAScript.cs:394]: ah I see you do

Comment (maridematte) [reply tests/Install-Scripts.Test/GivenThatIWantToGetTheSdkLinksFromAScript.cs:394]: Yes, I added on a recent commit

**Comment by YuliiaKovalova** [comment]
Looks good.
Please extend the documentation too: https://learn.microsoft.com/en-us/dotnet/core/tools/dotnet-install-script#:~:text=%2D%2D-,os,-%3COPERATING_SYSTEM%3E

**Review by YuliiaKovalova — APPROVED** [review]

**Pull request merged**